conmon-rs/server/src/streams.rs

```diff
@@ -93,34 +91,57 @@ impl Streams {
     fn read_loop_single_stream(
         message_tx: Sender<Message>,
         stop_rx: CrossbeamReceiver<()>,
-        stream: Stream,
+        fd: RawFd,
     ) -> Result<()> {
-        trace!("Start reading from single stream: {:?}", stream);
+        trace!("Start reading from single fd: {:?}", fd);
 
         let message_tx_clone = message_tx.clone();
-        thread::spawn(move || {
-            let mut reader = BufReader::new(stream);
-            let mut buf = vec![0; 1024];
-            loop {
-                match reader.read(&mut buf) {
+        let (thread_shutdown_tx, thread_shutdown_rx) = oneshot::channel();
+        thread::spawn(move || Self::create_tokio_runtime(message_tx_clone, fd, thread_shutdown_rx));
```

Comment:
Is there a way to reuse the original runtime?

Reply:
Wait wait, why are we creating multiple runtimes?  That seems like a bad idea.

Use e.g. https://docs.rs/tokio/latest/tokio/runtime/struct.Handle.html to reference the runtime.

(See an example usage in https://github.com/coreos/rpm-ostree/pull/3575 where we have legacy C/C++ code which is spawning threads, and we want to use Rust/Tokio from inside those threads; that's done by sharing the `Handle`).

Reply:
I don't understand why we're spawning a thread for this at all versus directly using `AsyncRead` and representing it as a future.

But if for some reason we need to use blocking IO from a thread, https://docs.rs/tokio/latest/tokio/task/fn.spawn_blocking.html is a more Tokio-native way to do that.
(We use that in ostree-rs-ext because we're using the `tar` crate which uses blocking IO.  Also see e.g. https://tokio.rs/tokio/topics/bridging )

Reply:
I was able to reuse the current runtime by relying only on tokio tasks. This seems to work, thank you for the input folks! :pray: